packages/lesswrong/lib/collections/users/karmaChangesGraphQL.js

```diff
@@ -1,58 +1,28 @@
 import Users from "meteor/vulcan:users";
 import { getKarmaChanges, getKarmaChangeDateRange, getKarmaChangeNextBatchDate } from "../../karmaChanges.js";
-import { addGraphQLSchema, addGraphQLResolvers, Connectors, getSetting } from 'meteor/vulcan:core';
-import { Comments } from '../comments'
-import { Posts } from '../posts'
+import { addGraphQLSchema, addGraphQLResolvers, getSetting } from 'meteor/vulcan:core';
 
-addGraphQLSchema(`
-  type PostWithScoreChange {
-    scoreChange: Int
-    post: Post
-  }
-`);
-addGraphQLSchema(`
-  type CommentWithScoreChange {
-    scoreChange: Int
-    comment: Comment
-  }
-`);
 addGraphQLSchema(`
   type KarmaChanges {
     totalChange: Int
     startDate: Date
     endDate: Date
     nextBatchDate: Date
     updateFrequency: String
-    posts: [PostWithScoreChange]
-    comments: [CommentWithScoreChange]
+    posts: JSON
+    comments: JSON
   }
 `);
 
 addGraphQLResolvers({
   KarmaChanges: {
     posts: async (karmaChangesJSON, args, context) => {
-      let posts = _.filter(karmaChangesJSON.documents, d=>d.collectionName==="Posts");
-      let postPromises = _.map(posts, post => Connectors.get(Posts, post._id));
-      let postObjects = await Promise.all(postPromises);
-      let postsById = {};
-      _.each(postObjects, post => postsById[post._id] = post);
-      
-      return _.map(posts, post => ({
-        scoreChange: post.scoreChange,
-        post: postsById[post._id],
-      }));
+      let postChanges = _.filter(karmaChangesJSON.documents, d=>d.collectionName==="Posts");
```

Comment:
I think we can remove these resolvers completely, and just get the correct document in the `karmaChangesInCollectionPipeline` (which also avoids the somewhat awkward passing of `documents`)